src/plugins/intel_cpu/src/nodes/concat.cpp

```diff
@@ -560,6 +593,112 @@ void Concat::execNspcSpecCase() {
     });
 }
 
+void Concat::execRef() {
+    const size_t numSrc = getParentEdges().size();
+    std::vector<const uint8_t*> srcPtrs;
+    const Memory& dstMemory = getChildEdgeAt(0)->getMemory();
+    const size_t elemSize = DnnlExtensionUtils::sizeOfDataType(dstMemory.GetDataType());
+    const auto& outputShape = dstMemory.getDescPtr()->as<BlockedMemoryDesc>()->getBlockDims();
+    uint8_t* dstPtr = reinterpret_cast<uint8_t*>(dstMemory.GetData());
+    for (size_t i = 0; i < numSrc; i++) {
+        const Memory& src_mem = getParentEdgesAtPort(i)[0]->getMemory();
+        srcPtrs.push_back(reinterpret_cast<const uint8_t*>(src_mem.GetData()) + srcsOffset[i] * elemSize);
```

Comment:
To avoid such calculations, one may use `Memory::GetPtr()` method, where the offset to the real data is applied. And please check other places, because technically offset may be in dstMemory as well.